auth0/src/main/java/com/auth0/android/request/internal/BaseRequest.kt

```diff
@@ -96,12 +96,18 @@ internal open class BaseRequest<T, U : Auth0Exception>(
         val reader = AwareInputStreamReader(response.body, Charset.defaultCharset())
         if (response.isSuccess()) {
             //2. Successful scenario. Response of type T
-            val result: T = resultAdapter.fromJson(reader)
+            val result: T = try {
+                resultAdapter.fromJson(reader)
+            } catch (exception: Exception) {
```

Comment:
The reason I opted to catch all exceptions here is because of the unexpected `JsonIOException` runtime exception, I documented in my https://github.com/google/gson/issues/1907 issue, which hides the checked `IOException` that `JsonAdapter.fromJson()` declares that it can throw.

Ideally I'd expect to catch both `IOException` as well as the `JsonParseException`s that `Streams.parse()` can throw, including `JsonIOException`, but [Kotlin's lack of multi catch](https://youtrack.jetbrains.com/issue/KT-7128) makes this more cumbersome. It makes sense to catch any exception that `JsonAdapter.fromJson()` can throw here, otherwise a bad JSON server response, including an `IOException` result, could result in a client crash.

Reply:
Ah, I see. I wasn't aware that that feature was missing in Kotlin. Thanks for the context. 

I agree those 3 exceptions are the ones we want to catch. Would you at least add a comment below the "catch" line mentioning the 3 exception classes we are intending to catch there? Perhaps in the future, this feature is available and we can use the multi-catch syntax.

Reply:
Done https://github.com/auth0/Auth0.Android/pull/486/commits/bf2ae839d44fd67a6586e61627d704b69ce390c6